ext/pdo_odbc/odbc_driver.c

```diff
@@ -344,6 +344,30 @@ static bool odbc_handle_set_attr(pdo_dbh_t *dbh, zend_long attr, zval *val)
 			}
 			H->assume_utf8 = bval;
 			return true;
+		case PDO_ATTR_AUTOCOMMIT:
+			if (!pdo_get_bool_param(&bval, val)) {
+				return false;
+			}
+			if (dbh->in_txn) {
+				pdo_raise_impl_error(dbh, NULL, "HY000", "Cannot change autocommit mode while a transaction is already open");
```

Comment:
This is consistent with firebird 's behaviour, but inconsistent with mysql's behaviour that just silently accepts it.
Bonus points: OCI actually commits the transaction when one is active.
So I'm not sure what the right behaviour should be.

Reply:
Firebird's autocommit mode is emulated by php, so it's implemented that way.
MySQL does not specifically commit with PDO, but it commits implicitly on the MySQL side.

I think there's room for debate here, but personally I think that from a user's point of view it's better to cause an error because if we "implicitly commit" it's hard to tell what happened.

Reply:
By the way, the ODBC does not define what should be the result of such an operation.

Therefore, the behavior is likely to differ depending on the DB, and at the very least, it is necessary to control the behavior on the pdo side, rather than "leaving it to the DB side" like MySQL.

Reply:
@nielsdos 

It would probably be better if the behavior here could be unified, so I'll post it to the mailing list. Maybe I should create an RFC if necessary.

It may be a good idea to leave this pull request on hold until the direction is decided.

Reply:
On the mailing list, there was an opinion that there is no need to unify this behavior in PDO, and that the behavior of each DB should be respected.

With ODBC, it is impossible to predict the type of DB that will be connected, so I personally think that a safe method is to generate an error all the time.

Reply:
Okay that's fair